src/lib/mdns/AllInterfacesListenIterator.h

```diff
@@ -0,0 +1,128 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <stdio.h>
+
+#include <mdns/minimal/Server.h>
+
+namespace chip {
+namespace Mdns {
+
+/// Checks if the current interface is powered on
+/// and not local loopback.
+template <typename T>
+bool IsCurrentInterfaceUsable(T & iterator)
```

Comment:
Yeah, totally one of my upcoming tasks - I didn't want to put it in with this PR just in case it broke something in the other code - easier to roll back like that.